backend/app/tasks/evaluate.py

```diff
@@ -0,0 +1,180 @@
+        evaluation_task_item.status = EvaluationStatus.ERROR
+
+        with Session(engine, expire_on_commit=False) as session:
+            session.merge(evaluation_task_item)
+            session.commit()
+
+
+def evaluate_task(evaluation_task_item: EvaluationTaskItem):
+    logger.info(f"Enter evaluate_task with evaluation item #{evaluation_task_item.id}")
+    ragas_list = [{
+        "user_input": evaluation_task_item.query,
+        "reference": evaluation_task_item.reference,
+        "response": evaluation_task_item.response,
+    }]
+    logger.debug(f"Response data {evaluation_task_item.response}")
+
+    ragas_dataset = EvaluationDataset.from_list(ragas_list)
+    logger.debug(f"Dataset {ragas_dataset.to_pandas().head()}")
+
+    evaluator_llm = LlamaIndexLLMWrapper(OpenAI(model="gpt-4o", api_key=settings.EVALUATION_OPENAI_API_KEY))
+    evaluator_embeddings = LlamaIndexEmbeddingsWrapper(OpenAIEmbedding(model="text-embedding-3-large", api_key=settings.EVALUATION_OPENAI_API_KEY))
+
+    metrics = [
+        # LLMContextRecall(llm=evaluator_llm),  # retrieved_contexts required
+        FactualCorrectness(llm=evaluator_llm),
+        # Faithfulness(llm=evaluator_llm),  # retrieved_contexts required
+        SemanticSimilarity(embeddings=evaluator_embeddings)
+    ]
+
+    try:
+        eval_result = evaluate(dataset=ragas_dataset, metrics=metrics, raise_exceptions=True, show_progress=False)
+
+        logger.debug(f"eval_result to_pandas")
+        result_list = eval_result.to_pandas().to_dict(orient='records')
+        logger.debug(f"result list {result_list}")
+        if len(result_list) != 1:
+            raise Exception(f"Item {evaluation_task_item.id} cannot get evaluation from ragas")
+
+        logger.debug(f"result {result_list[0]}")
+        evaluation_task_item.factual_correctness = result_list[0][FactualCorrectness.name]
+        evaluation_task_item.semantic_similarity = result_list[0][SemanticSimilarity.name]
+        evaluation_task_item.status = EvaluationStatus.DONE
+
+        logger.info(f"Result evaluation item #{evaluation_task_item}")
+        with Session(engine, expire_on_commit=False) as session:
+            session.merge(evaluation_task_item)
+            session.commit()
+    except Exception as e:
+        logger.error(f"Failed to evaluate item #{evaluation_task_item.id}, error: {e}")
+        evaluation_task_item.error_msg = traceback.format_exc()
+        evaluation_task_item.status = EvaluationStatus.ERROR
+
+        with Session(engine, expire_on_commit=False) as session:
+            session.merge(evaluation_task_item)
+            session.commit()
+
+
+@retry(stop=stop_after_attempt(2), wait=wait_fixed(5))
+def generate_answer_by_autoflow(messages: list, chat_engine: str) -> (str, str):
+    response = httpx.post(
+        url=settings.TIDB_AI_CHAT_ENDPOINT,
+        headers={
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {settings.TIDB_AI_API_KEY}",
```

Comment:
`TIDB_AI_API_KEY ` is a SecretStr, should use `settings.TIDB_AI_API_KEY.get_secret_value()`